File: backend/src/Operum.API/Controllers/AdminController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Operum.API.Controllers.Base;
using Operum.Service.Interfaces;

namespace Operum.API.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    [Authorize(Roles = "Admin")]
    public class AdminController(IAdminService adminService) : BaseController
    {
        [HttpGet("stats")]
        public async Task<IActionResult> GetStats()
        {
            return GetApiResponse(await adminService.GetStats());
        }

        [HttpGet("trackers")]
        public async Task<IActionResult> GetAllTrackers()
        {
            return GetApiResponse(await adminService.GetAllTrackers());
        }
    }
}

File: frontend/src/features/admin/api/adminController.ts
import api from "../../../shared/api/api";
import { ApiResponse } from "../../../shared/types/ApiResponse";
import { AdminStatsDto } from "../types/AdminStatsDto";
import { AdminTrackerDto } from "../types/AdminTrackerDto";

export const adminController = {
    getStats: async (): Promise<ApiResponse<AdminStatsDto>> => {
        return await api.get("/admin/stats");
    },
    getAllTrackers: async (): Promise<ApiResponse<AdminTrackerDto[]>> => {
        return await api.get("/admin/trackers");
    },
};
